### PR TITLE
Remove warning instance id ts

### PR DESCRIPTION
**Pull request by haakonvt**
Incorporate instance_id as an equal to id and external_id in the TimeSeries/Datapoints API.

https://cognitedata.atlassian.net/browse/DM-2128

## TODO
Some more tests, particularly on combining identifiers.

### CODE REVIEW

File: tests/tests_integration/test_api/test_datapoints.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def all_retrieve_endpoints(cognite_client, retrieve_endpoints):
 
 
 @pytest.fixture(scope="session")
-def instance_ts_id(cognite_client_alpha: CogniteClient, alpha_test_space: Space) -> InstanceId:
+def instance_ts_id(cognite_client: CogniteClient, alpha_test_space: Space) -> InstanceId:
```
Comment (doctrino) [inline tests/tests_integration/test_api/test_datapoints.py:187]: is the alpha client no longer used? Should we remove the fixture?

File: cognite/client/_api/datapoints.py [hunk 2]
```diff
@@ -536,13 +526,9 @@ def __init__(self, config: ClientConfig, api_version: str | None, cognite_client
...
         self,
         *,
-        id: None | int | DatapointsQuery | dict[str, Any] | Sequence[int | DatapointsQuery | dict[str, Any]] = None,
-        external_id: None
-        | str
-        | DatapointsQuery
-        | dict[str, Any]
-        | SequenceNotStr[str | DatapointsQuery | dict[str, Any]] = None,
-        instance_id: None | NodeId | Sequence[NodeId] = None,
+        id: None | int | DatapointsQuery | Sequence[int | DatapointsQuery] = None,
```
Comment (doctrino) [inline cognite/client/_api/datapoints.py:529]: isn't this breaking? not to allow `dict[str, Any]` any more?

**Review by doctrino — COMMENTED** [review]
Just some clarification.

File: tests/tests_integration/test_api/test_datapoints.py (hunk 1 above)
Comment (haakonvt) [reply tests/tests_integration/test_api/test_datapoints.py:187]: Still in use by files tests. Maybe these can also be changed. I decided not to remove it as we will certainly have alpha features in the futures

File: cognite/client/_api/datapoints.py (hunk 2 above)
Comment (haakonvt) [reply cognite/client/_api/datapoints.py:529]: Just the type annotations being updated to push people a little bit 😉

**Review by doctrino — APPROVED** [review]

**Pull request merged**